examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart

```diff
@@ -37,22 +37,20 @@ class _DatePickerExampleState extends State<DatePickerExample> {
   void _showDialog(Widget child) {
     showCupertinoModalPopup<void>(
       context: context,
-      builder: (BuildContext context) => Container(
-        height: 216,
-        padding: const EdgeInsets.only(top: 6.0),
-        // The Bottom margin is provided to align the popup above the system
-        // navigation bar.
-        margin: EdgeInsets.only(
-          bottom: MediaQuery.of(context).viewInsets.bottom,
-        ),
-        // Provide a background color for the popup.
-        color: CupertinoColors.systemBackground.resolveFrom(context),
-        // Use a SafeArea widget to avoid system overlaps.
-        child: SafeArea(
-          top: false,
-          child: child,
-        ),
-      ),
+      builder:
+          (BuildContext context) => Container(
+            height: 216,
+            padding: const EdgeInsets.only(top: 6.0),
+            // The Bottom margin is provided to align the popup above the system
+            // navigation bar.
+            margin: EdgeInsets.only(
+              bottom: MediaQuery.of(context).viewInsets.bottom,
+            ),
+            // Provide a background color for the popup.
+            color: CupertinoColors.systemBackground.resolveFrom(context),
+            // Use a SafeArea widget to avoid system overlaps.
+            child: SafeArea(top: false, child: child),
+          ),
```

Comment:
Why the extra line break after `builder:` and all this extra white space?

Reply:
This one is trickier. Sometimes it looks good to have the `(params...) => Function(` part of a closure on the same line as a surrounding named argument but often it looks worse. So currently the formatter doesn't have any special handling for it and falls back to the default rule that if the expression after the named argument splits then we split at the named argument too.

If you change the closure to use `{ ... }` instead of `=>`, you'll get better formatting. If you think the formatter should handle this differently, definitely file a bug and we can talk about it.

Reply:
Do you have an example where it looks worse? My intuition would have been the other way around.

Changing this to a block instead of `=>` makes formatting overall worse because of https://github.com/dart-lang/dart_style/issues/1527. It would also be sad if we couldn't use the compact `=>` shorthand anymore. With the current formatting, it feels as heavy as the block syntax and it kinda feels out of place in the argument list. I filed https://github.com/dart-lang/dart_style/issues/1545 to see if we can do something better here.

Reply:
> Do you have an example where it looks worse? My intuition would have been the other way around.

Not offhand, no, sorry. I'd have to give it a try and see if any of the regression tests get weird.

Thanks for filing the issue. Builder arguments are a good point and it's worth investigating whether there's something better the formatter can do here.